gcp/src/main/java/org/apache/iceberg/gcp/auth/GoogleAuthManager.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.gcp.auth;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.catalog.SessionCatalog;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.gcp.GCPProperties;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.rest.RESTClient;
+import org.apache.iceberg.rest.auth.AuthManager;
+import org.apache.iceberg.rest.auth.AuthSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An authentication manager that uses Google Credentials (typically Application Default
+ * Credentials) to create {@link GoogleAuthSession} instances.
+ *
+ * <p>This manager can be configured with properties such as:
+ *
+ * <ul>
+ *   <li>{@code gcp.auth.credentials-path}: Path to a service account JSON key file. If not set,
+ *       Application Default Credentials will be used.
+ *   <li>{@code gcp.auth.scopes}: Comma-separated list of OAuth scopes to request. Defaults to
+ *       "https://www.googleapis.com/auth/cloud-platform".
+ * </ul>
+ */
+public class GoogleAuthManager implements AuthManager {
+  private static final Logger LOG = LoggerFactory.getLogger(GoogleAuthManager.class);
+  public static final String DEFAULT_SCOPES = "https://www.googleapis.com/auth/cloud-platform";
+  private final String name;
+
+  private GoogleCredentials credentials;
+  private boolean initialized = false;
+
+  public GoogleAuthManager(String managerName) {
+    this.name = managerName;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  private void initialize(Map<String, String> properties) {
+    if (initialized) {
+      return;
+    }
+
+    String credentialsPath = properties.get(GCPProperties.GCP_CREDENTIALS_PATH_PROPERTY);
```

Comment:
I wonder if this will work with Authorization Code + `OAuth2RefreshCredentialsHandler`? The Auth Manager created by this component does not have access to an interactive terminal and a user, so an Authorization Code Flow would timeout and fail.

Reply:
Using the credentials from the attached service account is the preferred method for finding credentials in a production environment on Google Cloud. Many Google Cloud services let you attach a service account that can be used to provide credentials for accessing Google Cloud APIs. If ADC does not find credentials it can use in either the GOOGLE_APPLICATION_CREDENTIALS environment variable or the well-known location for local ADC credentials, it uses the [metadata server](https://cloud.google.com/compute/docs/metadata/overview) to get credentials for the service where the code is running.

When user uses GCP auth , it is not always Authorization Code flow. Is this answer your question ? @adutra

Reply:
Indirectly, yes. I suppose `OAuth2RefreshCredentialsHandler` would run on a host where the credentials used for authentication with the catalog server would use a regular `client_credentials` grant type.